src/Enzo/enzo_EnzoConfig.cpp

```diff
@@ -2031,12 +1986,295 @@ void EnzoConfig::read_physics_(Parameters * p)
+        double eta = eta_exists ? eta_list[0] : 0.001;
+        out = EnzoDualEnergyConfig::build_modern_formulation(eta);
+      } else if (type == "bryan95"){
+        ASSERT3("parse_de_config_",
+                "\"%s\" was used to specify %d value(s). When specified for "
+                "the \"%s\" dual energy formalism, it must provide 2 value.",
+                eta_paramname.c_str(), (int)names.size(), type.c_str(),
+                (eta_list.size() == 2) | !eta_exists);
+        double eta_1 = eta_exists ? eta_list[0] : 0.001;
+        double eta_2 = eta_exists ? eta_list[1] : 0.1;
+        out = EnzoDualEnergyConfig::build_bryan95_formulation(eta_1, eta_2);
+      } else {
+        ERROR1("parse_de_config_",
+               "\"Physics:fluid_props:dual_energy:type\" is invalid: \"%s\"",
+               type.c_str());
+      }
+    }
+
+    // look for dual energy parameters specified within the hydro solver (for
+    // backwards compatibility)
+    if (hydro_type != ""){
+      std::string legacy_de_param = "Method:" + hydro_type + ":dual_energy";
+      bool legacy_param_exists = p->param(legacy_de_param) != nullptr;
+
+      if (legacy_param_exists & !missing_de_config){
+        ERROR1("parse_de_config_",
+               "legacy parameter, \"%s\", duplicates other parameters",
+               legacy_de_param.c_str());
+      } else if (legacy_param_exists) {
+        WARNING1("parse_de_config_",
+                 "\"%s\" is a legacy parameter that will be removed",
+                 legacy_de_param.c_str());
+        bool use_de = p->value_logical(legacy_de_param, false);
+        if (!use_de){
+          out = EnzoDualEnergyConfig::build_disabled();
+        } else if (hydro_type == "ppm"){
+          out = EnzoDualEnergyConfig::build_bryan95_formulation
+            (p->value_float("Method:ppm:dual_energy_eta_1", 0.001),
+             p->value_float("Method:ppm:dual_energy_eta_2", 0.1));
+        } else {
+          out = EnzoDualEnergyConfig::build_modern_formulation
+            (p->value_float("Method:mhd_vlct:dual_energy_eta", 0.001));
+        }
+      }
+    }
+    return out;
+  }
+
+  //----------------------------------------------------------------------
+
+  EnzoFluidFloorConfig parse_fluid_floor_config_(Parameters * p,
+                                                 const std::string& hydro_type,
+                                                 bool using_grackle)
+  {
+    // initialize default values (a value <= 0 means there is no floor)
+    double density_floor = 0.0;
+    double pressure_floor = 0.0;
+    double temperature_floor = 0.0;
+    double metal_mass_frac_floor = 0.0;
+
+    auto get_ptr_to_floor_var = [&](const std::string name)
+      {
+        if (name == "density") { return &density_floor; }
+        if (name == "pressure") { return &pressure_floor; }
+        if (name == "temperature") { return &temperature_floor; }
+        if (name == "metalicity") { return &metal_mass_frac_floor; }
```

Comment:
```suggestion
        if (name == "metallicity") { return &metal_mass_frac_floor; }
```

Reply:
Thanks for catching this. I'll be sure to fix this!